### ISSUE TITLE
Compiler warnings in libqhull_r

### ISSUE THREAD

**Issue by davidcsterratt** [qhull/qhull#36]
i updating the R geometry library with the latest version of Qhull (pulled from github today), and I notice the following warnings when compiling:
```
d:/Compiler/gcc-4.9.3/mingw_32/bin/gcc  -I"D:/RCompile/recent/R-3.5.2/include" -DNDEBUG  -I"d:/RCompile/CRANpkg/lib/3.5/Rcpp/include" -I"d:/RCompile/CRANpkg/lib/3.5/RcppProgress/include"   -I"d:/Compiler/gcc-4.9.3/local330/include"  -include Rgeometry.h   -O3 -Wall  -std=gnu99 -mtune=core2 -c merge_r.c -o merge_r.o
merge_r.c: In function 'qh_premerge':
merge_r.c:79:7: warning: variable 'apexpointid' set but not used [-Wunused-but-set-variable]
   int apexpointid;
       ^
merge_r.c: In function 'qh_getpinchedmerges':
merge_r.c:1787:32: warning: unused variable 'opposite' [-Wunused-variable]
   vertexT *nearest, *pinched, *opposite= NULL, *bestvertex= NULL, *bestpinched= NULL;
                                ^
merge_r.c:1786:9: warning: unused variable 'ridges' [-Wunused-variable]
   setT *ridges= NULL;
         ^
merge_r.c: In function 'qh_merge_pinchedvertices':
merge_r.c:2568:9: warning: unused variable 'ridges' [-Wunused-variable]
   setT *ridges= NULL;
         ^
merge_r.c: In function 'qh_next_facetmerge':
merge_r.c:4019:10: warning: unused variable 'bestdist' [-Wunused-variable]
   coordT bestdist= REALmax;
          ^
merge_r.c: In function 'qh_rename_adjacentvertex':
merge_r.c:4390:9: warning: unused variable 'angledegen' [-Wunused-variable]
   realT angledegen= qh_ANGLEdegen;
         ^
d:/Compiler/gcc-4.9.3/mingw_32/bin/gcc  -I"D:/RCompile/recent/R-3.5.2/include" -DNDEBUG  -I"d:/RCompile/CRANpkg/lib/3.5/Rcpp/include" -I"d:/RCompile/CRANpkg/lib/3.5/RcppProgress/include"   -I"d:/Compiler/gcc-4.9.3/local330/include"  -include Rgeometry.h   -O3 -Wall  -std=gnu99 -mtune=core2 -c poly2_r.c -o poly2_r.o
poly2_r.c: In function 'qh_checkfacet':
poly2_r.c:823:11: warning: suggest parentheses around operand of '!' or change '|' to '||' or '!' to '~' [-Wparentheses]
       if (!ridge->tested | ridge->nonconvex | ridge->mergevertex) {
           ^
poly2_r.c: In function 'qh_matchdupridge':
poly2_r.c:2271:60: warning: unused variable 'samecycle_count' [-Wunused-variable]
   int skip, newskip, nextskip= 0, maxskip= 0, maxskip2= 0, samecycle_count= 0, makematch;
                                                            ^
poly2_r.c: In function 'qh_matchdupridge_coplanarhorizon':
poly2_r.c:2443:82: warning: unused variable 'ismergeridge' [-Wunused-variable]
   boolT same, ismatch, ismergehorizon, isbesthorizon= False, isfirstmatch= True, ismergeridge= False, isdup;
                                                                                  ^
poly2_r.c:2443:40: warning: unused variable 'isbesthorizon' [-Wunused-variable]
   boolT same, ismatch, ismergehorizon, isbesthorizon= False, isfirstmatch= True, ismergeridge= False, isdup;
                                        ^
poly2_r.c: In function 'qh_addfacetvertex':
poly2_r.c:47:5: warning: 'vertex_i' may be used uninitialized in this function [-Wmaybe-uninitialized]
     qh_setaddnth(qh, &facet->vertices, vertex_i, newvertex);
     ^
In file included from ./Rgeometry.h:24:0,
                 from <command-line>:0:
poly2_r.c: In function 'qh_matchdupridge_coplanarhorizon':
./qhull_ra.h:82:47: warning: 'horizon' may be used uninitialized in this function [-Wmaybe-uninitialized]
 #define trace2(args) {if (qh->IStracing >= 2) qh_fprintf args;}
                                               ^
poly2_r.c:2438:42: note: 'horizon' was declared here
   facetT *facet, *newfacet, *nextfacet, *horizon, *newhorizon, *neighbor;
                                          ^
poly2_r.c: In function 'qh_opposite_vertex':
poly2_r.c:2825:7: warning: 'facet' may be used uninitialized in this function [-Wmaybe-uninitialized]
       qh_errexit2(qh, qh_ERRqhull, facet, neighbor);
       ^
poly2_r.c: In function 'qh_replacefacetvertex':
poly2_r.c:3120:10: warning: 'vertex_n' may be used uninitialized in this function [-Wmaybe-uninitialized]
     new_i--;
          ^
```
It would be great if you could investigate.

**Comment by cbbarber** [comment]
Thanks David.   Developer release 2018.0.1 (7.3.0) was premature.    Besides the gcc warnings that you report, the code does not perform as well as the old code on some inputs.   In addition, when built under Gitbash on Windows, './qhull' and './qconvex' do not print the help message.  They wait for stdin instead.

Please use v7.2.1 for now.  It is the commit prior to 2018.0.1

If someone knows how to move 2018.0.1 to a git branch, please let me know.

**Comment by davidcsterratt** [comment]
I can think of how to create a stable branch, e.g. "v7.2" and leave the developer release on the master branch. I think it would be a case of:
```
git checkout tags/v7.2.1
git branch v7.2
git push origin v7.2
```
You could then commit any bug fixes to the v7.2 on that branch, and keep development on master.

**Comment by cbbarber** [comment]
Qhull 2019.1 compiles clean with many of the gcc warnings.   After the next update to github, review the warning list for CC_WARNINGS and CXX_WARNINGS in Makefile
